docs/enhancements/add-external-providers.md

```diff
@@ -0,0 +1,182 @@
+---
+title: Add-Providers-Support
+authors:
+  - "@Gal-Zaidman"
+creation-date: 2021-07-25
+last-updated: 2021-07-25
+---
+
+# Add Providers Support
+
+## Summary
+
+Currently, the assisted installer supports Bare-Metal installations only.
+Openshift can be installed on various providers such as:
+- Red Hat Virtualization.
+- OpenStack.
+- VSphere.
+- AWS.
+- GCP.
+- Azure.
```

Comment:
We don't intend on supporting cloud installations, just on-prem.

Reply:
Well yes but I'm just stating here that Openshift can be installed on them.
If you feel it should be mentioned I can add it into the Non-Goals, but since this enhancement is for adding the provider interface, add oVirt and migrate existing providers BM and vSphere I don't think it is required.

Reply:
Not sure what do you mean by migrate....

Reply:
Perhaps we can use some wording like:

"OpenShift has an interface to select the target provider, which is currently not being used by assisted-service (visit this link to learn more about OCP's provider support).

This enhancement will focus on exposing such interface through assisted-installer in order to make it easier to enable more providers in the future"

Probably needs some tweaks but you get the idea :)

Reply:
- @romfreiman I meant move the current parts of the code (for example vSphere support that was recently added) to the provider interface.

- @avishayt again since it's just the summery I have no strong feeling here, but I don't think it hurts stating that Opeshift supports also cloud providers... we are talking about a provider interface. I will add it as a non-goal to make it clearer.
- @flaper87 I think that will make it a lot more complex than it should be since provider support is much bigger than the installer, and the installer project doesn't really have such an interface :).

Reply:
The question "which platforms does assisted work on" has caused a lot of confusion across openshift engineering in the past. The specific question "but why would I want to use assisted on AWS" has definitely come up a lot, the answer was, or at least from what I understand, is that you wouldn't want to because we have IPI there and it works well for that scenario.

To reduce potential for confusion in this doc, I would recommend not listing the platforms we know RH doesn't intend to use assisted on, else you open yourself to those kinds of questions again

Reply:
OK edited